app/mm-redux/actions/threads.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ThreadTypes, PostTypes, UserTypes} from '@mm-redux/action_types';
+
+import {Client4} from '@client/rest';
+import {getCurrentUserId} from '@mm-redux/selectors/entities/common';
+import {getCurrentTeamId} from '@mm-redux/selectors/entities/teams';
+import {batchActions, DispatchFunc, GenericAction, GetStateFunc} from '@mm-redux/types/actions';
+import {UserThread, UserThreadList} from '@mm-redux/types/threads';
+
+import ThreadConstants from '../constants/threads';
+import {logError} from './errors';
+import {forceLogoutIfNecessary} from './helpers';
+import {getMissingProfilesByIds} from './users';
+
+export function getThreads(userId: string, teamId: string, before = '', after = '', perPage = ThreadConstants.THREADS_CHUNK_SIZE, deleted = false, unread = true, since = 0) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let userThreadList: UserThreadList;
+
+        try {
+            userThreadList = await Client4.getUserThreads(userId, teamId, before, after, perPage, true, deleted, unread, since);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        if (userThreadList) {
+            const getThreadsActions: Array<GenericAction> = [
+                {
+                    type: ThreadTypes.RECEIVED_THREADS,
+                    data: {
+                        ...userThreadList,
+                        threads: userThreadList?.threads?.map((thread) => ({...thread, is_following: true})) ?? [],
```

Comment:
`userThreadList` should be defined at this point.. is there a reason why it wouldn't have threads? 

does the deconstruction on the line above plays any part? seems to be overwritten with this and the lines below